popgen.md

```diff
@@ -0,0 +1,339 @@
+engine = stdpopsim.get_engine("msprime")
+ts = engine.simulate(model, contig, samples, seed=9)
+print(f"Simulated a tree sequence of {ts.num_samples} haploid genomes:")
+print(f"{ts.num_sites} variable sites over {ts.sequence_length/1e3} kb of sequence")
+```
+
+We can now inspect alleles and their frequencies at the variable sites we have simulated
+along the genome:
+
+```{code-cell}
+# Display a couple of variable sites (those in the first kilobase of sequence)
+for v in ts.variants(right=1_000):
+    display(v)
+```
+
+Or we can display the {meth}`~TreeSequence.haplotypes` (i.e. the variable sites) for
+each sample
+
+```{code-cell}
+samples = ts.samples()
+for sample_id, h in zip(samples, ts.haplotypes(samples=samples)):
+    pop = ts.node(sample_id).population
+    print(f"Sample {sample_id:<2} ({ts.population(pop).metadata['name']:^5}): {h}")
+```
+
+From the tree sequence it is easy to obtain the
+{meth}`TreeSequence.allele_frequency_spectrum` for the entire region (or for
+{ref}`windowed regions<sec_tskit_getting_started_compute_statistics_windowing>`)
+
+```{code-cell}
+afs = ts.allele_frequency_spectrum()
+plt.bar(range(ts.num_samples + 1), afs)
+plt.title("Allele frequency spectrum")
+plt.show()
+```
+
+Similarly `tskit` allows fast and easy
+{ref}`calculation of statistics<sec_tutorial_stats>` along the genome. Here is
+a plot of windowed $F_{st}$ between Africans and admixed Americans over this short
+region of chromosome:
+
+```{code-cell}
+# Define the samples between which Fst will be calculated
+pop_id = {p.metadata["name"]: p.id for p in ts.populations()}
+sample_sets=[ts.samples(pop_id["AFR"]), ts.samples(pop_id["ADMIX"])]
+
+# Do the calculation
+windows = list(range(0, int(ts.sequence_length + 1), 1_000))
+F_st = ts.Fst(sample_sets, windows=windows)
+
+# Plot
+plt.stairs(F_st, windows, baseline=None)
+plt.ylabel("AFR-ADMIX Fst")
+plt.xlabel("Genome position")
+plt.show()
+```
+
+Extracting the genetic tree at a specific genomic location is easy using `tskit`, which
+also provides methods to {ref}`plot<sec_tskit_viz>` these trees. Here we
+grab the tree at position 10kb, and colour the different populations by
+different colours, as described in the {ref}`viz tutorial<sec_tskit_viz_styling>`:
+
+```{code-cell}
+tree = ts.at(10_000)
+
+colours = dict(AFR="yellow", EUR="cyan", ASIA="green", ADMIX="red")
+styles = [
+    f".leaf.p{pop.id} > .sym {{fill: {colours[pop.metadata['name']]}}}"
+    for pop in ts.populations()
+]
+
+styles += [ # rotate the population labels, etc
+    ".leaf > .lab {text-anchor: start; transform: rotate(90deg) translate(6px)}",
+    ".leaf > .sym {stroke: black}"
+]
+
+labels = { # Label samples by population
+    u: ts.population(ts.node(u).population).metadata["name"].capitalize()
+    for u in ts.samples()
+}
+
+tree.draw_svg(
+    size=(800, 500),
+    # canvas_size=(800, 520),  # Make room for labels - only valid in tskit >= 0.5.4
```

Comment:
Let's file an issue to update this when 0.5.4 is released.

Reply:
Good point. https://github.com/tskit-dev/tutorials/issues/211